internal/tui/components/chat/messages/renderer.go

```diff
@@ -101,13 +104,48 @@ func (br baseRenderer) renderWithParams(v *toolCallCmp, toolName string, args []
 	if v.isNested {
 		return v.style().Render(header)
 	}
-	if res, done := earlyState(header, v); done {
-		return res
-	}
+
 	body := contentRenderer()
 	return joinHeaderBody(header, body)
 }
```

Comment:
critical: verify if and where the shouldShowContentForState() function should be called.